### PR TITLE
Fix SyftAPI

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Review by madhavajay — CHANGES_REQUESTED** [review]
Lets discuss

File: packages/syft/src/syft/core/node/common/node.py [hunk 1]
```diff
@@ -383,7 +384,8 @@ def get_metadata_for_client(self) -> Metadata:
             version=str(__version__),
         )
 
-    def get_api(self) -> SyftAPI:
+    def get_api(self, signing_key: Optional[SyftSigningKey] = None) -> SyftAPI:
```
Comment (rasswanth-s) [inline packages/syft/src/syft/core/node/common/node.py:387]: MAYBE: Lets make the signing_key a mandatory argument, so we could enforce them to always pass the signing key of the user requesting the api.?

File: packages/syft/src/syft/grid/client/grid_connection.py [hunk 2]
```diff
@@ -221,7 +221,6 @@ def _get_api(self, timeout: Optional[float] = 2) -> SyftAPI:
             print(f"Failed to upgrade to HTTPS. {e}")
 
         obj = _deserialize(response.content, from_bytes=True)
-        obj.api_url = f"{str(self.base_url)}/syft/new_api_call"
```
Comment (rasswanth-s) [inline packages/syft/src/syft/grid/client/grid_connection.py:224]: Q: Why have we removed it ?

File: packages/syft/src/syft/oblv/deployment_client.py [hunk 3]
```diff
@@ -291,7 +291,6 @@ def _get_api(self) -> SyftAPI:
         )
         self.sanity_check_oblv_response(req)
         obj = deserialize(req.content, from_bytes=True)
-        obj.api_url = f"{self.__conn_string}/worker/syft_api_call"
```
Comment (rasswanth-s) [inline packages/syft/src/syft/oblv/deployment_client.py:294]: Q: Why have we removed it?

File: packages/syft/src/syft/core/node/common/node.py [hunk 4]
```diff
@@ -383,7 +384,8 @@ def get_metadata_for_client(self) -> Metadata:
...
+        signing_key = self.signing_key if signing_key is None else signing_key
         return SyftAPI.for_user(node_uid=self.id, signing_key=self.signing_key)
```
Comment (rasswanth-s) [inline packages/syft/src/syft/core/node/common/node.py:389]: NITPICK : self.signing_key --> _signing_key

**Pull request closed without merging**